language/move-lang/src/parser/byte_string.rs

```diff
@@ -0,0 +1,304 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::errors::*;
+use byteorder::{BigEndian, ByteOrder};
+use std::convert::TryFrom;
+
+use crate::parser::syntax::make_loc;
+
+/// Escape sequence.
+enum Sequence {
+    /// \x00
+    Hex(String),
+    /// u{7FFF}
+    Unicode(String),
+    /// The start of an escape sequence. \
+    Generic,
+}
+
+#[derive(Default)]
+pub struct ByteStringCoder {
+    filename: &'static str,
+    start_offset: usize,
+    buffer: Vec<u8>,
+    escape_sequence: Option<Sequence>,
```

Comment:
I think this will make it harder to handle the hex sequence. And it might make it harder to add other sequences.

Reply:
I'm not sure how it will make it harder to do the hex sequence? I'm envisioning a separate loop for just the hex case, where it starts and stops at the { and } respectively

I don't see any other cases being added. And if they are, I don't see this solution being better than just a separate function or loop for that case